tests/conftest.py

```diff
@@ -57,6 +103,54 @@ def get_scf_electronic_band_gap_template(threshold_change: float = 1e-3) -> SCFO
     return scf_outputs
 
 
+def get_electronic_dos(
+    energy_points: List[int] = [-3, -2, -1, 0, 1, 2, 3],
+    total_dos: List[float] = [1.5, 1.2, 0, 0, 0, 0.8, 1.3],
+) -> ElectronicDensityOfStates:
+    """
+    Create an `ElectronicDensityOfStates` object with a template for the electronic_dos property. It uses
+    the template of the model_system created with the `get_model_system` function.
+    """
+    model_system = get_model_system()
+    variables_energy = [Energy(grid_points=energy_points * ureg.joule)]
+    electronic_dos = ElectronicDensityOfStates(variables=variables_energy)
+    electronic_dos.value = total_dos * ureg('1/joule')
+    orbital_s_Ga_pdos = DOSProfile(
+        name='orbital s Ga',
+        variables=variables_energy,
+        entity_ref=model_system.cell[0].atoms_state[0].orbitals_state[0],
+    )
+    orbital_px_As_pdos = DOSProfile(
+        name='orbital px As',
+        variables=variables_energy,
+        entity_ref=model_system.cell[0].atoms_state[1].orbitals_state[0],
+    )
+    orbital_py_As_pdos = DOSProfile(
+        name='orbital py As',
+        variables=variables_energy,
+        entity_ref=model_system.cell[0].atoms_state[1].orbitals_state[1],
+    )
+    orbital_s_Ga_pdos.value = [0.2, 0.5, 0, 0, 0, 0.0, 0.0] * ureg('1/joule')
+    orbital_px_As_pdos.value = [1.0, 0.2, 0, 0, 0, 0.3, 0.0] * ureg('1/joule')
+    orbital_py_As_pdos.value = [0.3, 0.5, 0, 0, 0, 0.5, 1.3] * ureg('1/joule')
+    electronic_dos.projected_dos = [
+        orbital_s_Ga_pdos,
+        orbital_px_As_pdos,
+        orbital_py_As_pdos,
+    ]
```

Comment:
This is a shame... I think one could set this up based on a model system input, together with a partial dos.

Reply:
Well, `entity_ref` works like that. The name was redundant, but I should add a method to extract the name (like I add as a TODO). Let me do it then.